### ISSUE TITLE
Size of feature_names error, should equal with total number of features

### ISSUE THREAD

**Issue by yangdi92** [Microsoft/LightGBM#540]
Please search your question on previous issues, stackoverflow or other search engines before you open a new one.

For bugs and unexpected issues, please provide following information, so that we could reproduce on our system.

## Environment info  
Operating System: win10 64bit
CPU: i5
R version: microsoft R 3.32
when i run the demo:

require(lightgbm)
require(methods)
data(agaricus.train, package = "lightgbm")
data(agaricus.test, package = "lightgbm")
train <- agaricus.train
test <- agaricus.test
class(train$label)
class(train$data)
print("Training lightgbm with sparseMatrix")
bst <- lightgbm(data = train$data,
                label = train$label,
                num_leaves = 4,
                learning_rate = 1,
                nrounds = 2,
objective = "binary")

## Error Message:
Error in lgb.call("LGBM_DatasetCreateFromCSC_R", ret = handle, private$raw_data@p,  : 
  api error: Size of feature_names error, should equal with total number of features

## i had succedd in runing this program ever, so, what is the problem? 
thank you!

**Comment by guolinke** [comment]
Just test it on latest code, and it runs successfully. 
Can you try re-install the latest code and restart R, then test again ?

**Comment by yangdi92** [comment]
Ok I wii try, thank you very much Sent from Mail Master
            On 05/22/2017 18:43, Guolin Ke wrote: Just test it on latest code, and it runs successfully.
Can you try re-install the latest code and restart R, then test again ?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Microsoft/LightGBM","title":"Microsoft/LightGBM","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Microsoft/LightGBM"}},"updates":{"snippets":[{"icon":"PERSON","message":"@guolinke in #540: Just test it on latest code, and it runs successfully. \r\nCan you try re-install the latest code and restart R, then test again ? "}],"action":{"name":"View Issue","url":"https://github.com/Microsoft/LightGBM/issues/540#issuecomment-303064854"}}}

**Comment by yangdi92** [comment]
I need to install version v2?Sent from Mail Master
            On 05/22/2017 18:43, Guolin Ke wrote: Just test it on latest code, and it runs successfully.
Can you try re-install the latest code and restart R, then test again ?

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Microsoft/LightGBM","title":"Microsoft/LightGBM","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Microsoft/LightGBM"}},"updates":{"snippets":[{"icon":"PERSON","message":"@guolinke in #540: Just test it on latest code, and it runs successfully. \r\nCan you try re-install the latest code and restart R, then test again ? "}],"action":{"name":"View Issue","url":"https://github.com/Microsoft/LightGBM/issues/540#issuecomment-303064854"}}}

**Comment by guolinke** [comment]
@yangdi92 the latest master branch.
